Core/src/org/sleuthkit/autopsy/modules/hashdatabase/HashDbManager.java

```diff
@@ -786,48 +1031,340 @@ public boolean equals(Object obj) {
+         *                of the case in which the content was encountered.
+         *
+         * @throws TskCoreException
+         */
+        @Override
+        public void addHashes(Content content, String comment) throws TskCoreException {
+            
+        }
+
+        /**
+         * Adds a list of hashes to the hash database at once
+         *
+         * @param hashes List of hashes
+         *
+         * @throws TskCoreException
+         */
+        @Override
+        public void addHashes(List<HashEntry> hashes) throws TskCoreException {
+            
+        }
+
+        /**
+         * Perform a basic boolean lookup of the file's hash.
+         *
+         * @param content
+         *
+         * @return True if file's MD5 is in the hash database
+         *
+         * @throws TskCoreException
+         */
+        @Override
+        public boolean lookupMD5Quick(Content content) throws TskCoreException {
+            // This only works for AbstractFiles and MD5 hashes 
+            assert content instanceof AbstractFile;
+            if (content instanceof AbstractFile) {
+                AbstractFile file = (AbstractFile) content;
+                if (null != file.getMd5Hash()) {
+                    try{
+                        return EamDb.getInstance().isFileHashInReferenceSet(file.getMd5Hash(), this.referenceSetID);
+                    } catch (EamDbException ex){
+                        Logger.getLogger(SleuthkitHashSet.class.getName()).log(Level.SEVERE, "Error performing central reposiotry hash lookup", ex); //NON-NLS
+                        throw new TskCoreException("Error performing central reposiotry hash lookup", ex);
+                    }
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Lookup hash value in DB and provide details on file.
+         *
+         * @param content
+         *
+         * @return null if file is not in database.
+         *
+         * @throws TskCoreException
+         */
+        @Override
+        public HashHitInfo lookupMD5(Content content) throws TskCoreException {
+            HashHitInfo result = null;
+            // This only works for AbstractFiles and MD5 hashes 
+            assert content instanceof AbstractFile;
+            if (content instanceof AbstractFile) {
+                AbstractFile file = (AbstractFile) content;
+                if (null != file.getMd5Hash()) {
+                    try{
+                        if(EamDb.getInstance().isFileHashInReferenceSet(file.getMd5Hash(), this.referenceSetID)){
+                            // Make a bare-bones HashHitInfo for now
+                            result = new HashHitInfo(file.getMd5Hash(), "", "");
+                        }
+                    } catch (EamDbException ex){
+                        Logger.getLogger(SleuthkitHashSet.class.getName()).log(Level.SEVERE, "Error performing central reposiotry hash lookup", ex); //NON-NLS
```

Comment:
I think it could help in debugging if we identify the database that experienced the error.